### PR TITLE
Allow testing old deps with postgres

### PR DESCRIPTION
**Pull request by erikjohnston**
This is so that we can test old dependencies with postgres

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -114,11 +114,17 @@ deps =
...
+setenv =
+    postgres: SYNAPSE_POSTGRES = 1
+
 commands =
     # Make all greater-thans equals so we test the oldest version of our direct
     # dependencies, but make the pyopenssl 17.0, which can work against an
     # OpenSSL 1.1 compiled cryptography (as older ones don't compile on Travis).
-    /bin/sh -c 'python -m synapse.python_dependencies | sed -e "s/>=/==/g" -e "s/psycopg2==2.6//" -e "s/pyopenssl==16.0.0/pyopenssl==17.0.0/" | xargs -d"\n" pip install'
+    #
+    # Also strip out psycopg2 unless we need it.
```
Comment (clokep) [inline tox.ini:125]: This is to ensure that we don't import it when it isn't available?

File: .buildkite/scripts/setup_old_deps.sh [hunk 2]
```diff
@@ -9,5 +9,3 @@ apt-get update
 apt-get install -y python3.5 python3.5-dev python3-pip libxml2-dev libxslt-dev xmlsec1 zlib1g-dev tox
 
 export LANG="C.UTF-8"
```
Comment (richvdh) [inline .buildkite/scripts/setup_old_deps.sh:11]: this export goes nowhere. I suspect it was here for a reason; you'll need to figure out if that reason was important.

File: .buildkite/scripts/setup_old_deps.sh [hunk 3]
```diff
@@ -9,5 +9,3 @@ apt-get update
...
-
-exec tox -e py35-old,combine
```
Comment (richvdh) [inline .buildkite/scripts/setup_old_deps.sh:13]: there are comments in the header which will need updating.

File: tox.ini [hunk 4]
```diff
@@ -114,11 +114,17 @@ deps =
     coverage
     coverage-enable-subprocess==1.0
 
+setenv =
+    postgres: SYNAPSE_POSTGRES = 1
+
```
Comment (richvdh) [inline tox.ini:119]: I believe this should be magically inherited from the defaults in `testenv`. could you verify one way or the other?

**Review by richvdh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**